contrib/drivers/hpe/nimble/client.go

```diff
@@ -0,0 +1,659 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nimble
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/astaxie/beego/httplib"
+	log "github.com/golang/glog"
+	. "github.com/opensds/opensds/contrib/drivers/utils/config"
+	"github.com/opensds/opensds/pkg/model"
+	pb "github.com/opensds/opensds/pkg/model/proto"
+)
+
+const (
+	requestRetryTimes            = 1
+	FcInitiatorDefaultGrpName    = "OsdsFcGrp"
+	IscsiInitiatorDefaultGrpName = "OsdsIscsiGrp"
+
+	tokenUrlPath         = "/tokens"
+	poolUrlPath          = "/pools"
+	poolDetailUrlPath    = "/pools/detail"
+	volumeUrlPath        = "/volumes"
+	snapshotUrlPath      = "/snapshots"
+	initiatorUrlPath     = "/initiators"
+	initiatorGrpUrlPath  = "/initiator_groups"
+	accessControlUrlPath = "/access_control_records"
+)
+
+// *****************Original Errors***********************
+func (e *ClinetErrors) Error() string {
+	var errStrings []string
+	for _, err := range e.Errs {
+		errStrings = append(errStrings, fmt.Sprint(err))
+	}
+	return fmt.Sprint(strings.Join(errStrings, "\n"))
+}
+
+func (e *ArrayInnerErrorBody) Error() string {
+	var errStrings []string
+	for _, err := range e.Errs {
+		errStrings = append(errStrings, fmt.Sprint(err.Error()))
+	}
+	return fmt.Sprint(strings.Join(errStrings, "\n"))
+}
+
+func (e *ArrayInnerErrorResp) Error() string {
+	return fmt.Sprintf("code:%v severity:%v text:%v", e.Code, e.Severity, e.Text)
+}
+
+// *******************************************************
+
+func unset(strings []string, search string) []string {
+	result := []string{}
+	deleteFlag := false
+	for _, v := range strings {
+		if v == search && !deleteFlag {
+			deleteFlag = true
+			continue
+		}
+		result = append(result, v)
+	}
+	return result
+}
+
+func NewClient(opt *AuthOptions) (*NimbleClient, error) {
+	endpoints := strings.Split(opt.Endpoints, ",")
+	c := &NimbleClient{
+		user:      opt.Username,
+		passwd:    opt.Password,
+		endpoints: endpoints,
+		insecure:  opt.Insecure,
+	}
+	err := c.login()
```

Comment:
Please handler this error first before return c.

Reply:
I added error handler before return c.